tests/unit/test_utils.py

```diff
@@ -2316,20 +2332,77 @@ def test_disabling_env_var_not_true(self):
         url = 'https://example.com/'
         env = {'AWS_EC2_METADATA_DISABLED': 'false'}
 
-        self.add_get_token_imds_response(token='token')
-        self.add_get_role_name_imds_response()
-        self.add_get_credentials_imds_response()
+        self.add_default_imds_responses()
 
         fetcher = InstanceMetadataFetcher(base_url=url, env=env)
         result = fetcher.retrieve_iam_role_credentials()
 
         self.assertEqual(result, self._expected_creds)
 
+    def test_IMDS_USE_IPV6(self):
```

Comment:
Since we've got a couple more changes, this should really be lowercase. It's generally against convention to use capitals in [functions](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names). The same would probably apply for `is_valid_IPv6_endpoint_url`